.github/workflows/clang_format.yml

```diff
@@ -0,0 +1,34 @@
+name: clang-format
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.x
+          architecture: x64
+      - name: Fetch PyTorch
+        uses: actions/checkout@v1
+      - name: Checkout PR tip
+        run: |
+          set -eux
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            # We are on a PR, so actions/checkout leaves us on a merge commit.
+            # Check out the actual tip of the branch.
+            git checkout ${{ github.event.pull_request.head.sha }}
+          fi
+          echo ::set-output name=commit_sha::$(git rev-parse HEAD)
+        id: get_pr_tip
+      - name: Run clang-format
+        run: |
+          set -eux
+          tools/clang_format.py --verbose --diff > ${GITHUB_WORKSPACE}/clang-format-output.txt
+          cat ${GITHUB_WORKSPACE}/clang-format-output.txt
```

Comment:
I think we should print a big obvious message at the top saying to run `python tools/clang_format.py` to fix this error.